### ISSUE TITLE
[Question] Combine early stopping and progress-independent LR adaptation

### ISSUE THREAD

**Issue by npit** [DLR-RM/stable-baselines3#1663]
### ❓ Question

I want to adapt the learning rate in a manner not dependent on the progress remaining, as done by [pytorch's ReduceLROnPlateau](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html). I want to use it alongside an early stopping callback, as an additional criterion. Pseudocode example of what I'm looking to achieve:

```python
# learning rate limits
lr, min_lr = 1e-1, 1e-6
# early stopping
es_counter, es_patience = 0, 5

while True:
   # train & validate
   get_rollout_and_train()
   val_results = run_validation()

   if improvement_exists(val_results):
      # doing fine, carry on
      continue

   es_counter += 1

   if lr > min_lr:
      # room to further decrease LR
      lr = max(lr/10, min_lr)
      es_counter = 0
      continue

   # end when no improvement is repeatedly observed over
   # the entire LR range
   if es_counter > es_patience:
      break
```

As far as I can see, I have to write a custom, single callback for the above functionality. Is that correct, or is there some existing functionality / tools that I'm missing?
Additionally, are there any difficulties / obstacles I'm disregarding? 
E.g. I I'll have to somehow hackily update the `lr_schedule` member variable of the model at each callback invocation, such that `BaseAlgorithm._update_learning_rate` works.

I'd appreciate any relevant feedback.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> As far as I can see, I have to write a custom, single callback for the above functionality.

yes, see https://github.com/DLR-RM/stable-baselines3/issues/735  (and others https://github.com/DLR-RM/stable-baselines3/issues/856 https://github.com/DLR-RM/stable-baselines3/issues/942 https://github.com/DLR-RM/stable-baselines3/issues/1336)

**Comment by npit** [comment]
Missed all these, thanks.
